control/control.go

```diff
@@ -530,101 +505,136 @@ func (p *pluginControl) validatePluginSubscription(pl core.SubscribedPlugin) []s
-		"namespace": mt.Namespace(),
-		"version":   mt.Version(),
+		"namespace": rmt.Namespace(),
+		"version":   rmt.Version(),
 	}).Info("subscription called on metric")
 
-	m, err := p.metricCatalog.Get(mt.Namespace(), mt.Version())
+	// match the requested metric with the cataloged metrics types
+	mts, err := p.metricCatalog.GetMatchedMetricTypes(rmt.Namespace(), rmt.Version())
 
 	if err != nil {
 		serrs = append(serrs, serror.New(err, map[string]interface{}{
-			"name":    mt.Namespace().String(),
-			"version": mt.Version(),
+			"name":    rmt.Namespace().String(),
+			"version": rmt.Version(),
 		}))
 		return serrs
 	}
 
-	// No metric found return error.
-	if m == nil {
-		serrs = append(serrs, serror.New(fmt.Errorf("no metric found cannot subscribe: (%s) version(%d)", mt.Namespace(), mt.Version())))
-		return serrs
+	if cdt == nil {
+		// initialize config tree if nil is provided
+		cdt = cdata.NewTree()
 	}
 
-	m.config = cd
+	for _, m := range mts {
+		config := cdt.Get(m.Namespace().Strings())
 
-	typ, serr := core.ToPluginType(m.Plugin.TypeName())
-	if serr != nil {
-		return []serror.SnapError{serror.New(err)}
-	}
+		if config == nil {
+			config = cdata.NewNode()
+		}
+		m.config = config
 
-	// merge global plugin config
-	if m.config != nil {
-		m.config.ReverseMerge(p.Config.Plugins.getPluginConfigDataNode(typ, m.Plugin.Name(), m.Plugin.Version()))
-	} else {
-		m.config = p.Config.Plugins.getPluginConfigDataNode(typ, m.Plugin.Name(), m.Plugin.Version())
-	}
+		typ, serr := core.ToPluginType(m.Plugin.TypeName())
+		if serr != nil {
+			return []serror.SnapError{serror.New(err)}
+		}
 
-	// When a metric is added to the MetricCatalog, the policy of rules defined by the plugin is added to the metric's policy.
-	// If no rules are defined for a metric, we set the metric's policy to an empty ConfigPolicyNode.
-	// Checking m.policy for nil will not work, we need to check if rules are nil.
-	if m.policy.HasRules() {
-		if m.Config() == nil {
-			serrs = append(serrs, serror.New(fmt.Errorf("Policy defined for metric, (%s) version (%d), but no config defined in manifest", mt.Namespace(), mt.Version())))
-			return serrs
+		// merge global plugin config
+		if m.config != nil {
+			m.config.ReverseMerge(p.Config.Plugins.getPluginConfigDataNode(typ, m.Plugin.Name(), m.Plugin.Version()))
+		} else {
+			m.config = p.Config.Plugins.getPluginConfigDataNode(typ, m.Plugin.Name(), m.Plugin.Version())
 		}
-		ncdTable, errs := m.policy.Process(m.Config().Table())
-		if errs != nil && errs.HasErrors() {
-			for _, e := range errs.Errors() {
-				serrs = append(serrs, serror.New(e))
+
+		// When a metric is added to the MetricCatalog, the policy of rules defined by the plugin is added to the metric's policy.
+		// If no rules are defined for a metric, we set the metric's policy to an empty ConfigPolicyNode.
+		// Checking m.policy for nil will not work, we need to check if rules are nil.
+		if m.policy.HasRules() {
```

Comment:
@candysmurf - I do not change this fragment of the code, it has been only moved into `for` loop.  But how it works is that if some rules are set in metric's policy, additional checks will be conducted - see it at master branch https://github.com/intelsdi-x/snap/blob/master/control/control.go#L574

So, I think it has to be kept in that form.